src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -925,9 +1049,12 @@ private void ParseHelper(string[] args)
                     {
                         break;
                     }
+
+                    ParametersUsed |= ParameterBitmap.SettingsFile;
                 }
                 else if (MatchSwitch(switchKey, "sta", "sta"))
                 {
+                    ParametersUsed |= ParameterBitmap.STA;
```

Comment:
This should be moved down before `_staMode = true;` to keep consistent with the rest -- collect info only if parsing is successful.